pkg/webui/account/store/middleware/user.js

```diff
@@ -12,28 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { createLogic } from 'redux-logic'
+import tts from '@account/api/tts'
 
-import api from '../../api'
-import * as user from '../actions/user'
+import api from '@account/api'
 
-const userLogic = createLogic({
+import createRequestLogic from '@ttn-lw/lib/store/logics/create-request-logic'
+
+import * as user from '@account/store/actions/user'
+
+import { selectUserId } from '@account/store/selectors/user'
+
+const logoutLogic = createRequestLogic({
   type: user.LOGOUT,
-  async process({ getState, action }, dispatch, done) {
-    try {
-      try {
-        await api.account.me()
-      } catch (error) {
-        dispatch(user.getUserMeFailure())
-      }
-      await api.account.logout()
-      dispatch(user.logoutSuccess())
-    } catch (error) {
-      dispatch(user.logoutFailure(error))
-    }
-
-    done()
+  process: async () => {
+    await api.account.logout()
+    window.location.reload()
+  },
+})
+
+const getUserLogic = createRequestLogic({
+  type: user.GET_USER,
+  process: async ({ action, getState }) => {
+    const {
+      meta: { selector },
+    } = action
+    const userId =
+      'payload' in action && action.payload.id ? action.payload.id : selectUserId(getState())
```

Comment:
Once the user is logged in, we can omit the user id and get it from the store. This does not work initially during the init logic. Since after the initialization the user is stored, there should be no situation where the user id is missing.